### PR TITLE
Copilot Chat: Include doc memory in upload/download a bot file

### PR DESCRIPTION
**Pull request by Vivihung**
### Motivation and Context
Include doc memory in upload/download a bot file.

### Description
1. Add a document embedding filed into the bot schema. 
2. Include doc memory when downloading a bot file. For user scope doc memory, the user object id will be replaced by chat id as a workaround. Eventually, we want to make doc memory to chat session scope.
3. Import the doc embedding as regular chat session embeddings. (We don't use the doc memory collection name when import. Things will get clear when we move doc memory to chat session scope.)

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] ~All unit tests pass, and I have added new tests where possible~
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs [hunk 1]
```diff
@@ -152,7 +156,7 @@ public class BotController : ControllerBase
     /// <param name="embeddingOptions">The embedding options.</param>
     /// <param name="botSchemaOptions">The bot schema options.</param>
     /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (Vivihung) [reply samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs:158]: I need a bot to check missing XML doc.... thanks for eyeballing them.

File: samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs [hunk 2]
```diff
@@ -114,29 +122,25 @@ public class BotController : ControllerBase
...
     /// <param name="kernel">The Semantic Kernel instance.</param>
     /// <param name="chatId">The chat id to be downloaded.</param>
-    /// <param name="aiServiceOptions">The AI service options.</param>
-    /// <param name="botSchemaOptions">The bot schema options.</param>
+    /// <param name="userId">The id of the current user and its home tenant.</param>
     /// <returns>The serialized Bot object of the chat id.</returns>
     [Authorize]
     [HttpGet]
-    [Route("bot/download/{chatId:guid}")]
+    [Route("bot/download/{chatId:guid}/{userId:regex([[a-b0-9]]+\\.[[a-b0-9]]+)}")]
```
[the remark this answers is not in the document]
Comment (glahaye) [reply samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs:129]: Could be useful: https://stackoverflow.com/questions/11040707/c-sharp-regex-for-guid

[the remark this answers is not in the document]
Comment (Vivihung) [reply samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs:129]: thanks for the ref @glahaye. We will soon remove userId from URL by fetching it from the auth token.

**Review by TaoChenOSU — APPROVED** [review]

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs [hunk 3]
```diff
@@ -16,28 +16,40 @@ namespace SemanticKernel.Service.Controllers;
...
+    private readonly AIServiceOptions _embeddingOptions;
+    private readonly DocumentMemoryOptions _documentMemoryOptions;
 
     /// <summary>
     /// The constructor of BotController.
     /// </summary>
     /// <param name="memoryStore">The memory store.</param>
     /// <param name="chatRepository">The chat session repository.</param>
     /// <param name="chatMessageRepository">The chat message repository.</param>
+    /// <param name="aiServiceOptions">The AI service options.</param>
```
[the remark this answers is not in the document]
Comment (Vivihung) [reply samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs]: added!

File: samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs [hunk 4]
```diff
@@ -57,19 +67,17 @@ public class BotController : ControllerBase
...
-        if (!this.IsBotCompatible(
+        if (!IsBotCompatible(
                 externalBotSchema: bot.Schema,
                 externalBotEmbeddingConfig: bot.EmbeddingConfigurations,
-                embeddingOptions: aiServiceOptions.Get(AIServiceOptions.EmbeddingPropertyName),
-                botSchemaOptions: botSchemaOptions.Value))
+                embeddingOptions: this._embeddingOptions,
+                botSchemaOptions: this._botSchemaOptions))
         {
             return this.BadRequest("Incompatible schema");
```
[the remark this answers is not in the document]
Comment (Vivihung) [reply samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs]: Updated!

**Review by adrianwyatt — APPROVED** [review]

**Pull request merged**